core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerAggregator.java

```diff
@@ -75,15 +75,16 @@ public void initialize(BatchRuntimeContext context) throws Exception {
     outputFields = conf.getOutputSchemaFields();
     outputMappings = conf.getFieldAliases();
     keyField = conf.getKeyField();
-    fieldName = conf.getFieldName();
-    fieldValue = conf.getFieldValue();
+    nameField = conf.getNameField();
+    valueField = conf.getValueField();
     outputSchema = initializeOutputSchema();
   }
 
   @Override
   public void groupBy(StructuredRecord record, Emitter<String> emitter) throws Exception {
     if (record.get(keyField) == null) {
-      if (!record.getSchema().getField(keyField).getSchema().isNullable()) {
+      if (record.getSchema().getField(keyField) != null && !record.getSchema().getField(keyField).getSchema()
```

Comment:
these conditions are different and should not result in the same error message. In the first case, the field is not in the schema at all. In the second, a non-nullable field has a null value.

The second doesn't actually need to be checked, as StructuredRecord.Builder does not let you set null for a non-nullable field.